executor/api/grpc/tensorflow/server_test.go

```diff
@@ -0,0 +1,209 @@
+package tensorflow
+
+import (
+	"context"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/onsi/gomega"
+	"github.com/seldonio/seldon-core/executor/api/client"
+	"github.com/seldonio/seldon-core/executor/api/grpc/seldon/proto"
+	"github.com/seldonio/seldon-core/executor/api/payload"
+	"github.com/seldonio/seldon-core/executor/proto/tensorflow/serving"
+	v1 "github.com/seldonio/seldon-core/operator/apis/machinelearning/v1"
+	"google.golang.org/grpc/metadata"
+	"io"
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+const (
+	TestMetaDataKey  = "foo"
+	TestMetaDataVal  = "bar"
+	TestModelVersion = int64(1)
+)
+
+type TestTensorflowClient struct {
+	t *testing.T
+}
+
+func NewTestTensorflowClient(t *testing.T) client.SeldonApiClient {
+	client := TestTensorflowClient{
+		t: t,
+	}
+	return &client
+}
+
+func (s TestTensorflowClient) Status(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	st := serving.GetModelStatusResponse{
+		ModelVersionStatus: []*serving.ModelVersionStatus{
+			&serving.ModelVersionStatus{
+				Version: TestModelVersion,
+			},
+		},
+	}
+	sm := payload.ProtoPayload{Msg: &st}
+	return &sm, nil
+}
+
+func (s TestTensorflowClient) Metadata(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.SeldonPayload, error) {
+	g := gomega.NewGomegaWithT(s.t)
+	md, ok := metadata.FromIncomingContext(ctx)
+	g.Expect(ok).NotTo(gomega.BeNil())
```

Comment:
Could dot import gomega so that you are able to use it without a qualifier. Tests using it read more easily, as per their examples:

Expect(ACTUAL).To(Equal(EXPECTED))